source/agora/tool/main.d

```diff
@@ -0,0 +1,98 @@
+/+
+ dub.sdl:
+ name "sometest"
+ dependency "vibe-d" version="~>0.8"
++/
+
+import agora.common.Types;
+import agora.api.FullNode;
+import agora.common.crypto.Key;
+import agora.common.Serializer;
+import agora.consensus.data.genesis.Test;
+import agora.consensus.data.Transaction;
+import agora.consensus.data.UTXOSetValue;
+import agora.consensus.UTXOSet;
+import agora.utils.Test;
+
+import std.algorithm;
+import std.random;
+import std.range;
+import std.stdio;
+
+import vibe.core.core;
+import vibe.d;
+
+final class MyUTXOSet : UTXOSet
+{
+    public this ()
+    {
+        super(":memory:");
+    }
+
+    public int opApply (scope int delegate (const Hash, const UTXOSetValue) dg)
+    {
+        auto results = this.utxo_db.db.execute("SELECT key, val FROM utxo_map");
+        foreach (row; results)
+        {
+            const key = deserializeFull!Hash(row.peek!(ubyte[])(0));
+            const value = deserializeFull!UTXOSetValue(row.peek!(ubyte[])(1));
+            if (auto ret = dg(key, value))
+                return ret;
+        }
+        return 0;
+    }
+}
+
+public int sendTransaction (API delegate (string address) api_maker)
+{
+    string address = "http://na-002.bosagora.io:7826";
+    auto node = api_maker(address);
+    logInfo("The address is %s", address);
+    const height = node.getBlockHeight();
+    logInfo("Node's block height: %s", height);
```

Comment:
Nice, we need to provide users with info on the current state of the chain 👍
I would however change the phrasing to something like "Initial block height for node is:" so that it's easier to grep if you later add more logging.